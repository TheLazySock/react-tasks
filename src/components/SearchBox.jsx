import React from 'react';

import BackdropContainer from '../components/BackdropContainer';

class SearchBox extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            text: this.props.search || '',
            searchBy: this.props.searchBy
        };
    }

    handleTextChange(e) {
        this.setState({
            text: e.target.value
        });
    }

    pressEnterKey(e) {
        if(e.keyCode == 13 && e.shiftKey == false) 
            this.handleSearch();
    }

    handleSearch() {
        const { text } = this.state;

        this.props.onSearch(text);
    };

    handleSearchTypeChanged(e) {
        this.setState({
            searchBy: e.target.value
        })

        this.handleSearchType(e.target.value);
    }

    handleSearchType(searchBy) {
        this.props.onSearchType(searchBy);
    }

    render() {
        const { text } = this.state;

        return (
            <div id="search-box">
                <h6>FIND YOUR MOVIE</h6>
                <input className="search-input" value={text} onChange={this.handleTextChange.bind(this)} placeholder="Search..." onKeyDown={this.pressEnterKey.bind(this)}/>
                <div className="search-box-elements">
                    <div className="search-by-box">
                        <label>SEARCH BY</label>
                        <button onClick={this.handleSearchTypeChanged.bind(this)} value="title" className={this.state.searchBy === "title" ? 'active' : null}>TITLE</button>
                        <button onClick={this.handleSearchTypeChanged.bind(this)} value="genres" className={this.state.searchBy === "genres" ? 'active' : null}>GENRE</button>
                    </div>
                    <button className={"search-button " + (!text ? "disabled" : "")} onClick={this.handleSearch.bind(this)} disabled={!text}>SEARCH</button>
                </div>
            </div>
        )
    };
}

export default SearchBox;