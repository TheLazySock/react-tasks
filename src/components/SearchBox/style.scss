@import "../../assets/constants.scss";

.search-box {
    display: flex;
    flex-direction: column;
    margin: auto;
    margin-top: 45px;
    width: 80%;
    & > h6 {
        font-size: 20px;
        color: #ffffff;
    }
    .search-input {
        color: #ffffff;
        background: #000000;
        min-height: 35px;
        padding: 0 15px;
        border: none;
        border-bottom: solid 2px $red;
        margin: 12px 0;
    }
}

.search-box-elements {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    .search-by-box {
        & > label {
            color: #ffffff;
            margin-right: 6px;
        }
        & > button {
            cursor: pointer;
            background-color: #8f8f8f;
            color: #ffffff;
            border: none;
            border-radius: 3px;
            // width: 50%;
            min-width: 45px;
            margin: 0 4px;
            padding: 3px 10px;
            &.active {
                background-color: $red;
            }
        }
        
    }
    .search-button {
        cursor: pointer;
        background-color: $red;
        color: #ffffff;
        border: none;
        border-radius: 3px;
        min-width: 140px;
        padding: 0 10px;
        &.disabled {
            background-color: #878787;
        }
    }
}