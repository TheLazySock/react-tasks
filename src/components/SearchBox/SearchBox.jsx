import React from 'react';
import style from './style.scss';

class SearchBox extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            text: this.props.search || '',
            searchBy: this.props.searchBy,
        };
    }

    handleTextChange(e) {
        this.setState({
            text: e.target.value,
        });
    }

    pressEnterKey(e) {
        if (e.keyCode === 13 && e.shiftKey === false) {
            this.handleSearch();
        }
    }

    handleSearch() {
        const { text } = this.state;

        this.props.onSearch(text);
    }

    handleSearchTypeChanged(e) {
        this.setState({
            searchBy: e.target.value,
        });

        this.handleSearchType(e.target.value);
    }

    handleSearchType(searchBy) {
        this.props.onSearchType(searchBy);
    }

    render() {
        const { text } = this.state;
        return (
            <div className={style['search-box']}>
                <h6>FIND YOUR MOVIE</h6>
                <input
                    className={style['search-input']}
                    value={text}
                    onChange={this.handleTextChange.bind(this)}
                    placeholder="Search..."
                    onKeyDown={this.pressEnterKey.bind(this)}
                />
                <div className={style['search-box-elements']}>
                    <div className={style['search-by-box']}>
                        <label>SEARCH BY</label>
                        <button
                            onClick={this.handleSearchTypeChanged.bind(this)}
                            value="title"
                            className={`
                                ${style['title-button ']} 
                                ${this.state.searchBy === 'title' ? style.active : null}`
                            }
                        >
                            TITLE
                        </button>
                        <button
                            onClick={this.handleSearchTypeChanged.bind(this)}
                            value="genres"
                            className={`
                                ${style['genres-button ']} 
                                ${this.state.searchBy === 'genres' ? style.active : null}`
                            }
                        >
                            GENRE
                        </button>
                    </div>
                    <button
                        className={`
                            ${style['search-button']} 
                            ${!text ? style.disabled : ''}`
                        }
                        onClick={this.handleSearch.bind(this)}
                        disabled={!text}
                    >
                        SEARCH
                    </button>
                </div>
            </div>
        );
    }
}

export default SearchBox;
