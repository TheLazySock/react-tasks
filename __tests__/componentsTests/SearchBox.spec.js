import React from 'react';
import renderer from 'react-test-renderer';
import SearchBox from '../../src/components/SearchBox';
import { shallow, configure, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('>>>SEARCH BOX tests', () => {
    let props = {
        search: 'pulp',
        searchBy: 'title',
        onSearch: jest.fn(),
        onSearchType: jest.fn()
    }
    let container;

    beforeEach(() => {
        container = mount(<SearchBox {...props} />);
    })

    it('search box snapshot', () => {
        expect(shallowToJson(container)).toMatchSnapshot();
    });

    it('state have to match props', () => {
        expect(container.instance().state.text).toBe(props.search);
        expect(container.instance().state.searchBy).toBe(props.searchBy);
    });

    it('trigger pressEnterKey', () => {
        let spy = spyOn(container.instance(), 'pressEnterKey');
        let trigger = container.find('input').simulate('keyDown', { key: 'Enter', keyCode: 13, which: 13, shiftKey: false });
        container.instance().pressEnterKey(trigger);
        expect(spy).toHaveBeenCalled();
    });

    it('trigger handleTextChange', () => {
        let spy = spyOn(container.instance(), 'handleTextChange');
        let trigger = container.find('input').simulate('change', { text: 'meow' });
        container.instance().handleTextChange(trigger);
        expect(spy).toHaveBeenCalled();
    });

    it('trigger handleSearch', () => {
        let spy = spyOn(container.instance(), 'handleSearch');
        let trigger = container.find('.search-button').simulate('click', { text: 'meow' });
        container.instance().handleSearch(trigger);
        expect(spy).toHaveBeenCalled();
    });

    it('trigger handleSearchTypeChanged', () => {
        let spy = spyOn(container.instance(), 'handleSearchTypeChanged');
        let trigger = container.find('.genres-button').simulate('click', { target: { value: 'genres' } });
        container.instance().handleSearchTypeChanged(trigger);
        expect(spy).toHaveBeenCalled();
    });
});