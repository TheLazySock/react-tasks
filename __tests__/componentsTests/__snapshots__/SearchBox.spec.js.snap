// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>SEARCH BOX tests search box snapshot 1`] = `
<SearchBox
  onSearch={[MockFunction]}
  onSearchType={[MockFunction]}
  search="pulp"
  searchBy="title"
>
  <div
    id="search-box"
  >
    <h6>
      FIND YOUR MOVIE
    </h6>
    <input
      className="search-input"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Search..."
      value="pulp"
    />
    <div
      className="search-box-elements"
    >
      <div
        className="search-by-box"
      >
        <label>
          SEARCH BY
        </label>
        <button
          className="title-button active"
          onClick={[Function]}
          value="title"
        >
          TITLE
        </button>
        <button
          className="genres-button null"
          onClick={[Function]}
          value="genres"
        >
          GENRE
        </button>
      </div>
      <button
        className="search-button "
        disabled={false}
        onClick={[Function]}
      >
        SEARCH
      </button>
    </div>
  </div>
</SearchBox>
`;
